PatientWebApplication/HealthClinic.DAL/Service/EmployeesScheduleService.cs

```diff
@@ -61,14 +61,16 @@ private Shift getScheduleShiftForDoctor(DoctorUser doctor, string date, Schedule
 
         public Shift getShiftForDoctorForSpecificDay(string date, DoctorUser doctor)
         {
+            if(doctor == null)
+            {
+                return null;
+            }
             List<Schedule> listOfSchedule = _employeesScheduleRepository.GetAll();
-
             foreach (Schedule schedule in listOfSchedule)
             {
                 Shift scheduleShiftForDoctor = getScheduleShiftForDoctor(doctor, date, schedule);
                 if (scheduleShiftForDoctor != null) return scheduleShiftForDoctor;
```

Comment:
Write this in form:
"if (getScheduleShiftForDoctor(doctor, date, schedule) != null) return getScheduleShiftForDoctor(doctor, date, schedule);"